File: src/main/java/com/micronaut/bug/config/SecurityContext.java
package com.micronaut.bug.config;

public class SecurityContext {

    private static final ThreadLocal<User> HOLDER = new ThreadLocal<>();

    public static void set(User user) {
        HOLDER.set(user);
    }

    public static User getUser() {
        return HOLDER.get();
    }

    public static void clear() {
        HOLDER.remove();
    }
}

File: src/main/kotlin/com/micronaut/bug/config/SecurityContext.kt
package com.micronaut.bug.config

object SecurityContext {

    private val HOLDER = ThreadLocal<User>()

    fun set(user: User) {
        HOLDER.set(user)
    }

    fun getUser(): User? {
        return HOLDER.get()
    }

    fun clear() {
        HOLDER.remove()
    }
}
